src/query/vacuum.c

```diff
@@ -1351,10 +1351,18 @@ vacuum_finalize (THREAD_ENTRY * thread_p)
 
   if (vacuum_Block_data_buffer != NULL)
     {
-      if (vacuum_Data.is_loaded && vacuum_consume_buffer_log_blocks (thread_p) != NO_ERROR)
+      if (vacuum_Data.is_loaded)
 	{
-	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_GENERIC_ERROR, 0);
-	  assert (0);
+	  while (!vacuum_Block_data_buffer->is_empty ())
```

Comment:
To avoid redundancy : Is it not better to remove "if (vacuum_Data.is_loaded)" and use only "while (!vacuum_Block_data_buffer->is_empty ()).."
and get rid by "if (!vacuum_Block_data_buffer->is_empty ())"?

Reply:
is rewrote code more clear?

Reply:
Yes.